boards/acd52832/src/main.rs

```diff
@@ -201,6 +202,7 @@ pub unsafe fn main() {
     //
     let button = components::button::ButtonComponent::new(
         board_kernel,
+        capsules::button::DRIVER_NUM as u32,
```

Comment:
Can we just decide on if these are u32 or usize? Then we don't need to have `as u32` everywhere.

Reply:
I feel like there was already some discussion on this based on the possibility of one day supporting 64-bit platforms, but I do not know what came of it

Reply:
I recall deciding that there are not going to be >= 2^32+1 drivers, so we can always say the driver num is 32 bits. But I don't know that for sure.

Reply:
> I recall deciding that there are not going to be >= 2^32+1 drivers, so we can always say the driver num is 32 bits. But I don't know that for sure.

I can confirm that. I have this on my todo list, but would rather not try to sneak this change in here (at least to existing code) as I feel it justifies a separate PR. That said, the new interfaces use `u32`s already, hence these conversions.

Reply:
I also would prefer not to block this PR on that change. It's been open for months and handling collisions with other PRs is difficult enough as-is.

Reply:
Then can we move the cast and have create grant take `usize`? If we don't do it now, it might not ever happen.

Reply:
Sure, but it means all the components that take the value as a u32 have to be changed to take a usize, only to be changed back in the next PR. I went ahead and did this though.